### PR TITLE
architecure: add a binaries section

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Add a section describing the various OP-TEE binaries produced by a build.

Suggested-by: Caleb Connolly <caleb.connolly@linaro.org>

### CODE REVIEW

File: architecture/porting_guidelines.rst [hunk 1]
```diff
@@ -207,6 +207,53 @@ typically will need to be shared with normal world, so there is need for some
...
+       be can executed from. Used by Hafnium to load OP-TEE, passed via
+       sp_layout.json.
+
+.. list-table:: Binaries generated only if given as targets to make
+   :header-rows: 1
+
+   * - Name
+     - Description
+   * - tee.srec
+     - Dump an srec file from tee-raw.bin using ``objcopy -I binary -O srec``
```
Comment (etienne-lms) [inline architecture/porting_guidelines.rst:251]: s/srec/SREC/

File: architecture/porting_guidelines.rst [hunk 2]
```diff
@@ -207,6 +207,53 @@ typically will need to be shared with normal world, so there is need for some
...
+   * - tee-pager.bin
+     - Legacy raw binary with pageable and init sections.
```
Comment (etienne-lms) [inline architecture/porting_guidelines.rst:253]: swap `pageable` and `pager` word at lines 253 and 255. Or maybe use `paged sections` and `unpaged` sections, to match to description of `tee-pager_v2.bin` ad `tee-pageable_v2.bin` above.
```
  * - tee-pager.bin
     - Legacy raw binary with unpaged and init sections.
   * - tee-pageable.bin
     - Legacy raw binary with paged sections.

File: architecture/porting_guidelines.rst [hunk 3]
```diff
@@ -207,6 +207,53 @@ typically will need to be shared with normal world, so there is need for some
...
+       by TF-A as bl32.bin to describe bl32_extra1.bin and bl32_extra2.bin.
+       Passed to TF-A with BL32=<filename>.bin when building the FIP.
+   * - tee-pager_v2.bin
+     - Raw binary for the unpaged and init parts of OP-TEE. Used by TF-A as
+       bl32_extra1.bin, requires tee-header_v2.bin. Passed to TF-A with
+       BL32_EXTRA1=<filename>.bin
+   * - tee-pageable_v2.bin
+     - Raw binary for the paged part of OP-TEE. Used by TF-A as
+       bl32_extra2.bin, requires tee-header_v2.bin. Passed to TF-A with
+       BL32_EXTRA1=<filename>.bin
```
Comment (etienne-lms) [inline architecture/porting_guidelines.rst:234]: `BL32_EXTRA2`

**Comment by jenswi-linaro** [comment]
Update

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` with a suggestion.

**Comment by jenswi-linaro** [comment]
Updated, squashed, and tag applied.

**Comment by jbech-linaro** [comment]
Thanks @jenswi-linaro and @etienne-lms , merging!

**Pull request merged**